src/ui.rs

```diff
@@ -150,12 +150,18 @@ impl Ui {
         }
 
         match self.output {
-            UiOutput::Paged { .. } => {}
-            UiOutput::Terminal { .. } => {
-                if io::stdout().is_tty() {
-                    self.output = UiOutput::new_paged_else_terminal(&self.settings);
+            UiOutput::Terminal { .. } if io::stdout().is_tty() => {
+                match UiOutput::new_paged(&self.settings) {
+                    Ok(new_output) => {
+                        self.output = new_output;
+                    }
+                    Err(e) => {
+                        self.write_warn(&format!("Failed to spawn pager: {}\n", e))
+                            .ok();
+                    }
```

Comment:
This orgnization makes sense. After the refactoring, I forgot that this was the only caller.